sale_order_type/models/sale.py

```diff
@@ -73,12 +74,27 @@ def onchange_type_id(self):
 
     @api.model
     def create(self, vals):
+        if self:
+            # This is a copy
+            # Give it a generic name
+            vals["name"] = "Copy of %s" % self[0].name
+            # Force user to select type_id, so proper
+            # sequencing will be maintained
+            vals["type_id"] = False
```

Comment:
This was to avoid changing the field definition, just in case copying is needed someplace else in the code.

Reply:
Well, but if people call copy someplace else in the code, `copy` in turn calls `create`, so they'll also run into the code here?

Reply:
Basically you are right, I should just set copy=False to type_id and be done with it.

Reply:
For usability it is much better to leave copy=True as is. Users are copying existing sale orders to re-use as much of the data from the existing record. That is including the type.

So I would go for a totally different approach. Clearly the `_("Copy of")` approach is valid for types that are not linked to a sequence. Otherwise get the next sequence for the type_id. Implement an on_change handler for type_id, and if the value really changes (compare with self._origin), get the next sequence if the new value also is linked to a sequence. If the old name was linked to a sequence, but a new one not, clear the name and force the user to enter a value.